Musee du Louvre.js

```diff
@@ -69,45 +67,81 @@ function doWeb(doc, url) {
 	if (detectWeb(doc, url) == "multiple") {
 		Zotero.selectItems(getSearchResults(doc, false), function (items) {
 			if (!items) {
-				return true;
+				return;
 			}
 			var articles = [];
 			for (var i in items) {
 				articles.push(i);
 			}
 			ZU.processDocuments(articles, scrape);
 		});
-	} else {
+	}
+	else {
 		scrape(doc, url);
 	}
 }
 
 
-function scrape(doc, url) {
-	var item = new Zotero.Item("artwork");
+function scrape(doc, _url) {
+	let item = new Zotero.Item('artwork');
+	let json = JSON.parse(text(doc, 'script[type="application/ld+json"]'));
 	
-	var box = doc.querySelector('.box-cartel ul');
-	var artist = text(box, 'li p', 0);
-	if (artist) {
-		artist = artist.replace(/\(.*\)/, '');
-		item.creators.push({
-			lastName: ZU.trimInternal(artist),
-			creatorType: "artist"
-		});
+	item.title = json.name || 'Untitled';
+	item.artworkMedium = json.material;
+	item.callNumber = json.identifier;
+	item.archive = "Louvre";
+	item.abstractNote = json.description;
+	item.url = json.url;
+	
+	if (json.creator) {
+		for (let artist of json.creator) {
+			// sometimes these aren't really artists ("School / Artistic centre"),
+			// but it's a royal pain to strip those out and there's nothing to
+			// clearly identify them
+			
+			let surnameMatch = artist.name.match(/^([^a-z\s.]{2,})\s*(.*)/);
+			if (surnameMatch) {
+				// if the artist's name starts with an uppercase surname
+				// (like "GOGH Vincent van") then pull that out.
+				
+				item.creators.push({
+					lastName: ZU.capitalizeTitle(surnameMatch[1], true),
+					firstName: surnameMatch[2],
+					creatorType: 'artist'
+				});
+			}
+			else {
+				item.creators.push(ZU.cleanAuthor(
+					artist.name,
+					'artist',
+					artist.name.includes(', ')
+				));
+			}
+		}
 	}
-	item.title = text(doc, 'h1>span');
-	if (!item.title) {
-		item.title = text(box, 'li p', 1);
+	
+	if (json.dateCreated) {
+		let bceDateMatch = json.dateCreated.match(/([0-9]+) av\./);
+		if (bceDateMatch) {
+			// https://docs.citationstyles.org/en/1.0.1/specification.html#ad-and-bc
+			item.date = '-' + bceDateMatch[1];
```

Comment:
@adam3smith is this the right approach for BCE dates? CSL recommends it, but I'm not having a lot of success generating actual BCE citations in Zotero.

Reply:
I'm not sure what citeproc-js supports currently, but we'll be supporting EDTF both in Zotero and in citeproc-rs in the not-too-distant future, so that's theoretically what we'd want to use. I believe that EDTF does use `-` for BCE dates ([as in ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Years)), but depending on the source data, the year may need to be shifted by one — e.g., `-0379` is 380 BCE.

Reply:
citeproc/CSL does support the - sign for BCE, but Zotero currently strips it out in date parsing, so it will only work in Extra (`issued: -379`). Not sure what you want to do about this while the current Zotero behavior exists. Seems weird to write this into Extra, so probably just leave and accept that it won't work in citation styles until the EDTF update?

Reply:
Oh, we can probably just fix that now (at least for the beta). [Issue](https://github.com/zotero/zotero/issues/2086) created.